addons/rating/models/rating.py

```diff
@@ -89,13 +89,45 @@ def _compute_rating_text(self):
     def create(self, values):
         if values.get('res_model_id') and values.get('res_id'):
             values.update(self._find_parent_data(values))
-        return super(Rating, self).create(values)
+        res = super(Rating, self).create(values)
+        if res.res_model and res.res_id:
+            res._compute_rating(res._group_model_ids())
+        return res
 
     @api.multi
     def write(self, values):
+        model_id_dict = {}
         if values.get('res_model_id') and values.get('res_id'):
             values.update(self._find_parent_data(values))
-        return super(Rating, self).write(values)
+        if values.get('res_model_id') or values.get('res_id'):
+            self._group_model_ids(model_id_dict)
+        res = super(Rating, self).write(values)
+        if model_id_dict != {} or values.get('rating') or values.get('consumed'):
+            self._group_model_ids(model_id_dict)
+            self._compute_rating(model_id_dict)
+        return res
+
+    @api.multi
+    def unlink(self):
+        model_id_dict = self._group_model_ids()
+        res = super(Rating, self).unlink()
+        self._compute_rating(model_id_dict)
+        return res
+
+    def _compute_rating(self, model_id_dict):
+        for res_model in model_id_dict:
+            items = self.env[res_model].browse(model_id_dict[res_model])
+            items.modified({'rating_ids'})
+            items.recompute()
+
+    def _group_model_ids(self, model_id_dict={}):
+        for record in self:
+            if record.res_model and record.res_id:
+                if record.res_model in model_id_dict:
```

Comment:
All the if/else structure could be replaced by a single line to add/append if you use a `list` instead of a `dict`.
Something like
```python
for record in self.filtered(lambda r: r.res_model and r.res_id):
    model_id_dict.setdefault(record.res_model, []).append(record.res_id)
return model_id_dict
```
instead of
```python
for record in self:
    if record.res_model and record.res_id:
        if record.res_model in model_id_dict:
            model_id_dict[record.res_model].add(record.res_id)
        else:
            model_id_dict[record.res_model] = {record.res_id}
return model_id_dict
```

Reply:
It is not a `dict` but a `set`. Since in the write method I have to run `group_model_ids` two times in case the res_model changes, I don't want to have duplicated ids in my `list` so I used a `set`.

I can do this: 
```python
for record in self.filtered(lambda r: r.res_model and r.res_id):
    model_id_dict.setdefault(record.res_model, set()).add(record.res_id)
return model_id_dict
```

Reply:
Sure :+1: